### PR TITLE
refactor default model

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Part of the major test refactoring

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by williamFalcon** [comment]
@Borda rebase pls

File: tests/callbacks/test_callbacks.py [hunk 1]
```diff
@@ -214,8 +214,6 @@ def on_test_end(self, trainer, pl_module):
 
 def test_early_stopping_no_val_step(tmpdir):
     """Test that early stopping callback falls back to training metrics when no validation defined."""
-    tutils.reset_seed()
-
     class ModelWithoutValStep(LightTrainDataloader, TestModelBase):
```
Comment (williamFalcon) [inline tests/callbacks/test_callbacks.py:217]: @Borda shouldn't this be replaced by the new test model and set val_step=None?

Comment (Borda) [reply tests/callbacks/test_callbacks.py:217]: in this one I have just replaced the `tutils.default_model`, the others are coming in followup PRs as I wanted to keep it per blocks not all at once

**Comment by Borda** [comment]
@williamFalcon rebased and waiting for tests to pass...

File: tests/callbacks/test_callbacks.py [hunk 2]
```diff
@@ -224,8 +222,7 @@ def training_step(self, *args, **kwargs):
             output.update({'my_train_metric': loss})
             return output
 
-    hparams = tutils.get_default_hparams()
-    model = ModelWithoutValStep(hparams)
+    model = ModelWithoutValStep(tutils.get_default_hparams())
```
Comment (williamFalcon) [inline tests/callbacks/test_callbacks.py:225]: @Borda we need to get rid of these models. the only one to use is the template model...

File: tests/loggers/test_neptune.py [hunk 3]
```diff
@@ -61,10 +61,7 @@ def test_neptune_additional_methods(neptune):
 
 def test_neptune_leave_open_experiment_after_fit(tmpdir):
     """Verify that neptune experiment was closed after training"""
-    tutils.reset_seed()
-
-    hparams = tutils.get_default_hparams()
-    model = LightningTestModel(hparams)
+    model = LightningTestModel(tutils.get_default_hparams())
```
Comment (williamFalcon) [inline tests/loggers/test_neptune.py:64]: wrong model

File: tests/loggers/test_trains.py [hunk 4]
```diff
@@ -8,8 +8,6 @@
 
 def test_trains_logger(tmpdir):
     """Verify that basic functionality of TRAINS logger works."""
-    tutils.reset_seed()
-
     hparams = tutils.get_default_hparams()
     model = LightningTestModel(hparams)
```
Comment (williamFalcon) [inline tests/loggers/test_trains.py:12]: wrong model

File: tests/models/test_restore.py [hunk 5]
```diff
@@ -74,8 +72,6 @@ def test_running_test_pretrained_model_distrib(tmpdir, backend):
 
 def test_running_test_pretrained_model_cpu(tmpdir):
     """Verify test() on pretrained model."""
-    tutils.reset_seed()
-
     hparams = tutils.get_default_hparams()
     model = LightningTestModel(hparams)
```
Comment (williamFalcon) [inline tests/models/test_restore.py:76]: wrong models

File: tests/trainer/test_dataloaders.py [hunk 6]
```diff
@@ -149,7 +146,6 @@ class CurrentTestModel(LightTrainDataloader, TestModelBase):
 
 def test_train_val_dataloaders_passed_to_fit(tmpdir):
     """ Verify that train & val dataloader can be passed to fit """
-    tutils.reset_seed()
```
Comment (williamFalcon) [inline tests/trainer/test_dataloaders.py:149]: we need seeds on top of every test... why are you removing them?

File: tests/trainer/test_optimizers.py [hunk 7]
```diff
@@ -94,7 +92,6 @@ class CurrentTestModel(
 
 
 def test_multi_optimizer_with_scheduling_stepping(tmpdir):
-    tutils.reset_seed()
 
     class CurrentTestModel(
             LightTestOptimizersWithMixedSchedulingMixin,
```
Comment (williamFalcon) [inline tests/trainer/test_optimizers.py:97]: wrong model

File: tests/trainer/test_trainer.py [hunk 8]
```diff
@@ -291,7 +287,6 @@ def mock_save_function(filepath):
 
 
 def test_model_freeze_unfreeze():
-    tutils.reset_seed()
 
     hparams = tutils.get_default_hparams()
     model = LightningTestModel(hparams)
```
Comment (williamFalcon) [inline tests/trainer/test_trainer.py:292]: wrong model

**Comment by Borda** [comment]
@williamFalcon this is only about default model, the others you commented are in other PRs

File: tests/trainer/test_dataloaders.py (hunk 6 above)
Comment (Borda) [reply tests/trainer/test_dataloaders.py:149]: Here the seed is is the function run...

File: tests/callbacks/test_callbacks.py (hunk 2 above)
Comment (Borda) [reply tests/callbacks/test_callbacks.py:225]: This is refactor only for default, the others are in other PRs per test package

**Pull request merged**